webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/policies/write/PolicyWriterService.scala

```diff
@@ -263,6 +264,39 @@ class PolicyWriterServiceImpl(
 
     //interpret HookReturnCode as a Box
 
+    def cleanPromiseFolderOfDeletedNode : Box[Seq[BetterFile]] = {
+      import BetterFile._
+      def getNodeFolders(file: BetterFile): List[BetterFile] = {
+        file.collectChildren(d => d.isDirectory && d.children.exists(_.name == "rules")).flatMap {
```

Comment:
I am *really* scared by the cost of this search with 10 000 nodes; doing it at each policy generation will slow it down even further

Reply:
Myabe it could be a batch that runs, every ... day ?

Reply:
(mean an independant system that runs outside policy generation)

Reply:
The cost should be really small. It's just collecting a biles names/properties. But OK to add a dedicated timer on that. 
It's not logical at all to make it run once in a while if the goal is to limit the timespan where non relevant files are present (if it's just space-consumption control, why not). 
And if can be put out of the generation itself, asynchronuously triggered so that it does not delay it.

Reply:
(there is no reason file operation like that should be slow when nothing is deleted, and if they are, we need to understand why, because we have a problem)

Reply:
It does not use much resource, apart lots of fd in an older version of better-files but fixed by updating it to latest version